srml/support/src/traits.rs

```diff
@@ -146,17 +148,24 @@ pub trait Currency<AccountId> {
 	/// be returned to notify of this. This is different behaviour to `unreserve`.
 	fn reserve(who: &AccountId, value: Self::Balance) -> result::Result<(), &'static str>;
 
-	/// Moves up to `value` from reserved balance to balance. This function cannot fail.
+	/// Moves up to `value` from reserved balance to free balance. This function cannot fail.
+	///
+	/// As much funds up to `value` will be moved as possible. If the reserve balance of `who`
+	/// is less than `value`, then `Some(remaining)` will be returned. If all of `value` is
+	/// moved, the function will return `None`.
 	///
-	/// As much funds up to `value` will be deducted as possible. If this is less than `value`,
-	/// then `Some(remaining)` will be returned. Full completion is given by `None`.
 	/// NOTE: This is different to `reserve`.
+	///
+	/// NOTE: If the remaining reserved balance is less than `ExistentialDeposit`, it will
```

Comment:
Two notes in the comment. Make `Note` a heading and put these notes as listings below it.